nlp/src/main/java/smile/nlp/dictionary/SimpleDictionary.java

```diff
@@ -44,9 +45,11 @@ public SimpleDictionary(String resource) {
         dict = new HashSet<>();
 
         File file = new File(resource);
-        try (BufferedReader input = file.exists() ?
-             new BufferedReader(new FileReader(resource)) :
-             new BufferedReader(new InputStreamReader(this.getClass().getResourceAsStream(resource)))) {
+        try (FileReader fileReader = file.exists() ? new FileReader(resource) : null;
+                InputStream resourceAsStream = this.getClass().getResourceAsStream(resource);
+                InputStreamReader inputStreamReader = new InputStreamReader(resourceAsStream);
+                BufferedReader input = file.exists() ? new BufferedReader(fileReader) :
+                        new BufferedReader(inputStreamReader)) {
```

Comment:
why do we change here? The original version is already using try with resources, right? I think the reader will close the underlying inputstream, is the fortify check complaining about the inputstream wrapped by the reader not being closed?

Besides the original version only creates inputStream-/Reader if the file does not exist, while the changed version will create it regardless.

Reply:
Yes, fortify check complains if we dont have separate resource declaration. Also, I think only the declared resources will be deleted